src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -269,10 +296,22 @@ private static <E extends Enum<E>> Object convertEnumConstants(MethodHandles.Loo
         }
     }
 
-    private static int doEnumSwitch(Enum<?> target, int startIndex, Object[] labels) {
+    private static int doEnumSwitch(Enum<?> target, int startIndex, Object[] unresolvedLabels,
+                                    MethodHandles.Lookup lookup, Class<?> enumClass,
+                                    ResolvedEnumLabels resolvedLabels) {
+        Objects.checkIndex(startIndex, unresolvedLabels.length + 1);
+
         if (target == null)
             return -1;
 
+        if (resolvedLabels.resolvedLabels == null) {
+            resolvedLabels.resolvedLabels = Stream.of(unresolvedLabels)
+                                                  .map(l -> convertEnumConstants(lookup, enumClass, l))
+                                                  .toArray();
+        }
+
+        Object[] labels = resolvedLabels.resolvedLabels;
+
         // Dumbest possible strategy
```

Comment:
I guess that given that we can't make any assumption about order here, this is the only possible strategy so I guess it makes sense to remove the comment

Reply:
I see, interesting